deeprank2/molstruct/atom.py

```diff
@@ -50,21 +49,19 @@ def __init__( # pylint: disable=too-many-arguments
         self._occupancy = occupancy
 
     def __eq__(self, other) -> bool:
-        return (
-            isinstance(self, type(other))
-            and self._residue == other._residue
-            and self._name == other._name
-        )
+        if isinstance (other, Atom):
+            return (self._residue == other._residue
+                    and self._name == other._name)
+        return NotImplemented
```

Comment:
We're not checking type hints with a tool (like mypy), but I guess that having here `bool` would give errors with the `NotImplemented` case

Reply:
I found this here, when researching something else (https://stackoverflow.com/questions/63503512/python-type-hinting-own-class-in-method). Not sure where I can find the formal correct way to format `__eq__`, but the commentors on this post all seem to agree that this is the correct way to do it.

Reply:
You could run [mypy](https://realpython.com/lessons/type-checking-mypy/) to verify that the type hints are right